roles/load_aware_undeploy_trimaran/tasks/main.yml

```diff
@@ -0,0 +1,35 @@
+- name: Ensure Trimaran is deployed
+  command: oc get ns trimaran --ignore-not-found -oname
+  register: has_trimaran
+
+- name: Delete Trimaran resources
+  command: 
+    oc delete -f {{ trimaran_setup_config }} --ignore-not-found
+  when: has_trimaran.stdout | length > 0
+
+- name: Ensure user applications monitor config is deployed
+  command: oc get configmap cluster-monitoring-config --ignore-not-found -oname -n openshift-monitoring
+  register: has_user_monitoring
+
+- name: Delete user monitoring configmap
+  command:
+    oc delete -f {{ user_applications_monitor_config }} --ignore-not-found
+  when: has_user_monitoring | length > 0
+
+- name: Wait for trimaran ns to disappear
+  shell: |
+    oc get ns trimaran --no-headers 2> /dev/null
+  register: has_trimaran_namespace
+  until: has_trimaran_namespace.rc != 0
+  failed_when: has_trimaran_namespace.rc == 0
+  delay: 5
+  retries: 30
+
+- name: Wait for monitoring configmap to disappear
+  shell: |
+    oc get configmap cluster-monitoring-config --no-headers 2> /dev/null
+  register: has_user_monitoring_deleted
+  until: has_user_monitoring_deleted.rc != 0
+  failed_when: has_user_monitoring_deleted.rc == 0
+  delay: 5
+  retries: 30
```

Comment:
this isn't necessary, when you do
```
oc delete <resource>
```
the command will block until the resource is destroyed (except is you pass `--wait=false`)

the same applied to the namespace deletion above actually